### ISSUE TITLE
Add sample data collection with Cloud Optimized GeoTIFF

### ISSUE THREAD

**Issue by j08lue** [EOEPCA/data-access#107]
We added a nice Sentinel-2 L2A data collection (over Poland), copying items from the Copernicus Data Space Ecosystem STAC that link to JPEG2000 Sentinel-2 band files:

- https://github.com/EOEPCA/data-access/issues/75

However, the JPEG2000 driver linked in GDAL (OpenJPEG) is failing under load and generally has poor performance

- https://github.com/EOEPCA/data-access/issues/105

So we need a nice collection instead that has Cloud Optimized GeoTIFF (COG).

To allow us to test and demo typical data access patterns, the collection should
1. Feature adjacent tiles / images that can be stitched together in dynamic tiling (mosaic) or coverages requests
2. Feature repeated coverage of the same area, such that we can test time series functionality

## Acceptance criteria

- [ ] eoAPI STAC in EOEPCA has a collection of COGs that meet the above criteria

**Comment by j08lue** [comment]
One possibility could be to copy selected collections from [Maxar Open Data](https://stacindex.org/catalogs/maxar-open-data-catalog-ard-format) to a CloudFerro S3 bucket and copy STAC records into our service.

**Comment by j08lue** [comment]
Maybe @MathewNWSH knows by heart, whether any nice COG collections exist in eodata.cloudferro.com that perhaps also already have STAC records somewhere?

I [previously](https://github.com/EOEPCA/data-access/issues/75#issuecomment-2324946487) noted that

Looking at the CloudFerro STAC https://radiantearth.github.io/stac-browser/#/external/https://pgstac.demo.cloudferro.com - the only collection with COGs seems to be [Sentinel-1 Ground Range Detected (GRD)](https://radiantearth.github.io/stac-browser/#/external/pgstac.demo.cloudferro.com/collections/sentinel-1-grd).

But maybe there are more?

**Comment by j08lue** [comment]
Sentinel-1 GRD could be an option, though. RGB composites of the different polarizations tend to look quite nice https://gis.stackexchange.com/a/400780

![image](https://github.com/user-attachments/assets/acd10da4-7793-40d1-8df1-dab5af1d2c3b)

**Comment by pantierra** [comment]
What about using Sentinel-2 COGs by [Amazon's Registry for Open Data](https://registry.opendata.aws/sentinel-2-l2a-cogs/). AFAIK there are no egress costs. And we could go for a similar area as before (e.g. Poland) to keep size of data a bit limited.

**Comment by pantierra** [comment]
Thanks, @MathewNWSH! @j08lue, I would prefer to use S2L2A over S1GRD, it is probably nicer to show. I can imagine it would be good to have this as a use case also when running it on other premises than cloudferro. 

What do you think for this purpose to download a timerange of a year or so of S2L2A-cogs over Poland (or even a smaller area) and the create a STAC Catalog from it (`rio-stac` should help us with that) that we can then ingest?

**Comment by j08lue** [comment]
Ok, let us copy some S2L2A COGs over to CloudFerro for now.

It would be great to have rich Sentinel-2 STAC metadata, more than what `rio-stac` produces.

I see three ways to get there:

1. Amend the [existing sentinel-2-l2a collection](https://radiantearth.github.io/stac-browser/#/external/eoapi.develop.eoepca.org/stac/collections/sentinel-2-l2a/items/S2B_MSIL2A_20231231T100329_N0510_R122_T33UYU_20231231T110801?.asset=asset-B04_10m) and just add another asset like `B04_10m_cog` for all bands, in original resolution (not the up/downsampled ones)
2. Copy STAC metadata from somewhere and change it to point to our bucket
3. Generate rich STAC data with https://github.com/stactools-packages/sentinel2

Could (1) be easiest, perhaps? We probably need to run one job per item to copy the relevant files.

**Comment by j08lue** [comment]
We need to find out how to create and get write access to a bucket in CloudFerro in the same region as our EOEPCA infrastructure. I would ask about that on EOEPCA Slack.

**Comment by pantierra** [comment]
We received access to the cloudferro infrastructure of the project. Next steps ahead.

* [ ] Download Sentinel-2 COG data for 2023 over the area of Iceland.
* [ ] Make an own STAC collection of this data (probably can just be taken and adapted from the search result of the STAC search of Earth Search)
* [ ] Spinn up a bucket for data storage on cloudferro through the infrastructure as a code
* [ ] Upload data and collection info to this bucket
* [ ] Link it to eoAPI deployed for EOEPCA+

**Comment by pantierra** [comment]
* There is [a script](https://github.com/pantierra/spaghetti/blob/main/stac_prepare_iceland.py) we use to obtain search results from Earth Search, strip out the `jp2` assets and leave only the `cogs` and dump this into a json file.
* The result is a [static catalog here](https://radiantearth.github.io/stac-browser/#/external/raw.githubusercontent.com/pantierra/static-stac/refs/heads/main/sentinel2-l2a-iceland/sentinel-2-l2a/collection.json?.language=en), and can be inspected with [this stac-browser](https://radiantearth.github.io/stac-browser/#/external/raw.githubusercontent.com/pantierra/static-stac/refs/heads/main/sentinel2-l2a-iceland/sentinel-2-l2a/collection.json?.language=en).
* It looks like we can use the files directly from the AWS Open Data Registry as they are publicly available. I suggest we get it to work first, and then consider whether we want to have a mirror on cloudferro.

 Next steps:
* Looking into ingestioning this into pgstac on the deployed eoapi.

**Comment by j08lue** [comment]
The sample looks great!

Agreed, if performance is ok with loading directly from AWS `us-west-2`, we could skip the copying. How many GB are we talking about, roughly?

**Comment by pantierra** [comment]
I currently took all imagery over Iceland from 2023 with a cloud coverage lower than 5%. There are 226 items of it. Each contain assets of around 180MB which should be a total of ~40GB. If we play with the cloud coverage it could be more or less.

**Comment by pantierra** [comment]
[Data should now be available](https://radiantearth.github.io/stac-browser/#/external/eoapi.develop.eoepca.org/stac/collections/sentinel-2-iceland?.language=en) - please check it @j08lue.

Related PRs:

* https://github.com/developmentseed/eoapi-k8s/pull/164 - simple ingestion method
* https://github.com/vincentsarago/MAXAR_opendata_to_pgstac/pull/9 - script to produce the pgstac import json files

**Comment by j08lue** [comment]
The collection looks great. As icing on the cake, we could add a collection thumbnail, so it looks nice in STAC Browser, similar to what the Poland Sentinel-2 collection has:

https://radiantearth.github.io/stac-browser/#/external/eoapi.develop.eoepca.org/stac/collections/sentinel-2-l2a?.asset=asset-thumbnail

```json
"assets": {
"thumbnail": {
"href": "https://s3.waw3-2.cloudferro.com/swift/v1/stac-png/S2_L2A.jpg",
"type": "image/jpeg",
"roles": [
"thumbnail"
],
"title": "Sentinel-2 Level-2A",
"proj:code": null,
"proj:shape": [
360,
640
]
}
},
```

Ideally a mosaic of the actual data or just one of the nicest scenes. Or just a screen grab from a cloudless Sentinel-2 product, like https://s2maps.eu/.

![image](https://github.com/user-attachments/assets/2faa6d28-e4f9-4839-beee-35b6a2328964)

**Comment by j08lue** [comment]
Done: https://radiantearth.github.io/stac-browser/#/external/eoapi.develop.eoepca.org/stac/collections/sentinel-2-iceland

**Comment by pantierra** [comment]
Yeah, once the Transaction endpoints work, i wanted to add the thumbnail through them as the first use.
